tools/cldr-apps/js/src/esm/cldrDash.js

```diff
@@ -0,0 +1,220 @@
+        if (newIssues.includes(issue)) {
+          updateNotification(data, xpath, issue, row);
+        } else {
+          removeNotification(data, xpath, issue);
+        }
+      });
+      newIssues.forEach((issue) => {
+        if (!oldIssues.includes(issue)) {
+          addNotification(data, xpath, issue, row);
+        }
+      });
+    });
+  } catch (error) {
+    console.error("Caught error in updateRow: " + error);
+  }
+}
+
+function updateNotification(data, xpath, issue, row) {
+  // TODO: modify the existing item in-place; that will be  more efficient, and will
+  // help to keep the items in their proper order, which is currently determined by the server
+  removeNotification(data, xpath, issue);
+  addNotification(data, xpath, issue, row);
+}
+
+function removeNotification(data, xpath, issue) {
+  for (let j in data.notifications) {
+    const n = data.notifications[j];
+    if (issue === n.notification) {
+      for (let g in n.entries) {
+        const entries = n.entries[g].entries;
+        for (let k in entries) {
+          if (entries[k].xpath === xpath) {
+            entries.splice(k, 1);
+            if (--n.total === 0) {
+              delete data.notifications[j];
+            }
+            delete xpathIndex[xpath][issue];
+            return;
+          }
+        }
+      }
+    }
+  }
+}
+
+function addNotification(data, xpath, issue, row) {
+  const n = getNotificationCategory(data, issue);
+  let comment = "";
+  row.items[row.winningVhash].tests.forEach((t) => {
+    if (t.type === issue) {
+      comment += t.message + "<br />";
+    }
+  });
+  const e = {
+    xpath: xpath,
+    code: row.code,
+    winning: row.winningValue,
+    english: row.displayName,
+    comment: comment,
+  };
+  const ph = getPathHeader(xpath);
+  const a = {
+    header: ph.header,
+    page: ph.page,
+    section: ph.section,
+    entries: [e],
+  };
+  n.entries.push(a);
+  n.total++;
+  console.log("addNotification added " + issue + ", new total = " + n.total);
+  if (!(xpath in xpathIndex)) {
+    xpathIndex[xpath] = {};
+  }
+  xpathIndex[xpath][issue] = e;
+}
+
+function getNotificationCategory(data, issue) {
+  const notifications = data.notifications;
+  let i = null;
+  for (let j in notifications) {
+    if (issue === notifications[j].notification) {
+      i = j;
+      break;
+    }
+  }
+  if (i === null) {
+    const obj = {
+      notification: issue,
+      total: 0,
+      entries: [],
+    };
+    notifications.push(obj); // TODO: insert in a particular order?
+    i = notifications.length - 1;
+  }
+  return notifications[i];
+}
+
+function getPathHeader(xpstrid) {
+  const xpathMap = cldrSurvey.getXpathMap();
+  const result = xpathMap.getImmediately({
+    hex: xpstrid,
+  });
+  if (result && result.ph) {
+    // success
+    return result.ph;
+  }
+  // failure; fall back on dummy header, and current section/page
+  return {
+    header: "[header]",
+    page: cldrStatus.getCurrentPage(),
```

Comment:
Log err?

Reply:
Good idea. I'll add that. The fallback might be superfluous... This all needs more testing...